pages/instructorAssessmentManualGrading/instanceQuestion/instanceQuestion.js

```diff
@@ -79,13 +86,78 @@ router.get(
   })
 );
 
+router.get(
+  '/grading_panel',
```

Comment:
Very minor nit: consider renaming to reflect that this isn't just the grading panel?

Reply:
@nwalters512 taking your brain here: the reason I moved to a separate path here was because middlewares that contain some of the data needed in the partials ran before the rubric update, so are out-of-date. This redirection forces the re-running of the middleware. However, the issue I'm having in this particular case now is that the CSRF-token doesn't match since it's generated with a different URL, so attempting to change the settings after this redirection causes a CSRF fail. So I need a different approach altogether. Here are some ideas:

* Instead of redirecting to this new URL, return the data in the POST code itself. This is often used in other contexts, and would likely be my preferred option if possible, since it is also one less RTT for redirection. The problem is that this would mean some of the locals values used in these EJS files would be out-of-date, since the middleware that updated them would have run before the updates, so it would require one of:
    * re-running the middlewares that are affected by the result;
    * updating the libs to re-select the associated values;
    * something else?
* replace this path with a query argument in the regular get, so the originalUrl is still the main page;
* re-compute the csrf-token in here with a custom URL (I don't really like this approach).

Do you (or @mwest1066) have a better idea on how to approach this?

Reply:
Handled by saving the CSRF token before submit and restoring it after the panels are replaced.